ui/core/src/model/units/time.ts

```diff
@@ -74,13 +74,73 @@ export enum PersesTimeToIntlTime {
   Years = 'year',
 }
 
+/**
+ * Note: This conversion will not be exactly accurate for months and years,
+ * due variations in the lengths of months (i.e. 28 - 31 days) and years (i.e. leap years).
+ * For precision with months and years, we would need more complex algorithms and/or external libraries.
+ * However, we expect that measurements in months and years will be rare.
+ */
+const TIME_UNITS_IN_SECONDS: Record<TimeUnitKind, number> = {
+  Years: 31536000, // 365 days
+  Months: 2592000, // 30 days
+  Weeks: 604800,
+  Days: 86400,
+  Hours: 3600,
+  Minutes: 60,
+  Seconds: 1,
+  Milliseconds: 0.001,
+};
+
+const LARGEST_TO_SMALLEST_TIME_UNITS: TimeUnitKind[] = [
+  'Years',
+  'Months',
+  'Weeks',
+  'Days',
+  'Hours',
+  'Minutes',
+  'Seconds',
+  'Milliseconds',
+];
+
+/**
+ * Choose the first time unit that produces a number greater than 1, starting from the biggest time unit.
+ */
+function getValueAndKindForNaturalNumbers(value: number, kind: TimeUnitKind): { value: number; kind: TimeUnitKind } {
+  const valueInSeconds = value * TIME_UNITS_IN_SECONDS[kind];
+
+  // Initialize for TS
+  const largestTimeUnit = LARGEST_TO_SMALLEST_TIME_UNITS[0] || 'Years';
+  let timeUnit: TimeUnitKind = largestTimeUnit;
+  let valueInTimeUnit: number = valueInSeconds / TIME_UNITS_IN_SECONDS[largestTimeUnit];
+
+  for (timeUnit of LARGEST_TO_SMALLEST_TIME_UNITS) {
+    valueInTimeUnit = valueInSeconds / TIME_UNITS_IN_SECONDS[timeUnit];
+    if (valueInTimeUnit >= 1) {
+      return { value: valueInTimeUnit, kind: timeUnit };
+    }
+  }
+
+  // If we didn't find a time unit, we have to settle for the smallest time unit (which is the last time unit).
+  return { value: valueInTimeUnit, kind: timeUnit };
+}
+
+function isMonthOrYear(kind: TimeUnitKind): boolean {
+  return kind === 'Months' || kind === 'Years';
+}
+
 export function formatTime(value: number, { kind, decimal_places }: TimeUnitOptions): string {
-  const isMonthOrYear = kind === 'Months' || kind === 'Years';
+  const { value: valueForNaturalNumbers, kind: kindForNaturalNumbers } = getValueAndKindForNaturalNumbers(value, kind);
+
+  return _formatTime(valueForNaturalNumbers, { kind: kindForNaturalNumbers, decimal_places });
+}
+
+function _formatTime(value: number, { kind, decimal_places }: TimeUnitOptions): string {
```

Comment:
Do you find this code ^ more readable @juliepagano 

I'm playing around, so open to yays or nays

Reply:
(I basically created a private internal function so we don't have to reassign `value` and `kind` inside of that function, `value` and `kind` are already determined by the time we get to this function)

Reply:
Seems ok to me, although I could see being a little confused about the extraction of some of the code into `_formatTime` in the future when I've forgotten this discussion. I was trying to think of a name for `_formatTime` that would communicate the remaining work it was doing, but couldn't think of anything helpful. If you can think of anything more descriptive, that might be helpful, but otherwise I think it's fine as is.